tests/system/test_system.py

```diff
@@ -552,6 +559,44 @@ def _unit_of_work(transaction, test):
             rows = list(after.execute_sql(self.SQL))
         self._check_rows_data(rows)
 
+    @unittest.skipIf(
+        USE_EMULATOR, "Request tagging feature is not supported by the emulator"
+    )
+    def test_db_run_in_transaction_with_request_options_then_check_request_tags(self,):
+        retry = RetryInstanceState(_has_all_ddl)
+        retry(self._db.reload)()
+
+        with self._db.batch() as batch:
+            batch.delete(self.TABLE, self.ALL)
+
+        def _unit_of_work(transaction, test):
+            rows = list(transaction.read(test.TABLE, test.COLUMNS, self.ALL))
+            test.assertEqual(rows, [])
+
+            transaction.insert_or_update(test.TABLE, test.COLUMNS, test.ROW_DATA)
+
+        self._db.run_in_transaction(_unit_of_work, test=self)
+
+        request_tag = "req-tag" + unique_resource_id("-")
+        with self._db.snapshot() as before:
+            rows = list(
+                before.execute_sql(
+                    self.SQL, request_options={"request_tag": request_tag}
+                )
+            )
+        self._check_rows_data(rows)
+        time.sleep(200)
```

Comment:
nit: could you add a fix-me / todo to rewrite this, since it will potentially delay the test?

One thing we could do is test if the request tag is there every `10` seconds and timeout after `200` seconds. This way the test can finish sooner.

Reply:
for now added a TODO.